stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt

```diff
@@ -130,6 +137,16 @@ public class ChannelListViewModel(
                     }
                 }
         }
+
+        if (ToggleService.isEnabled(ToggleService.TOGGLE_KEY_OFFLINE)) {
+            val queryChannelsRequest = QueryChannelsRequest(
+                filter = queryConfig.value.filters,
+                querySort = queryConfig.value.querySort,
+                limit = 30,
+                messageLimit = 1
+            )
+            queryChannelsState = chatClient.asReferenced().queryChannels(queryChannelsRequest).asState(viewModelScope)
```

Comment:
Why do we use `asReferenced()`?

Reply:
`asreferenced()` provides a wrapper around `ChatClient` that wraps requests in `QueryReference`. `QueryReference` allows us to access the offline state of the query.